streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -0,0 +1,319 @@
+
+        try {
+            produceSynchronously(
+                input,
+                asList(
+                    new KVT<>("k1", "v1", scaledTime(0L)),
+                    new KVT<>("k1", "v2", scaledTime(1L)),
+                    new KVT<>("k2", "v1", scaledTime(2L)),
+                    new KVT<>("x", "x", scaledTime(4L))
+                )
+            );
+            verifyOutput(
+                outputRaw,
+                asList(
+                    new KVT<>("v1", 1L, scaledTime(0L)),
+                    new KVT<>("v1", 0L, scaledTime(1L)),
+                    new KVT<>("v2", 1L, scaledTime(1L)),
+                    new KVT<>("v1", 1L, scaledTime(2L)),
+                    new KVT<>("x", 1L, scaledTime(4L))
+                )
+            );
+            verifyOutput(
+                outputSuppressed,
+                asList(
+                    new KVT<>("v1", 1L, scaledTime(0L)),
+                    new KVT<>("v1", 0L, scaledTime(1L)),
+                    new KVT<>("v2", 1L, scaledTime(1L)),
+                    new KVT<>("v1", 1L, scaledTime(2L)),
+                    new KVT<>("x", 1L, scaledTime(4L))
+                )
+            );
+        } finally {
+            driver.close();
+            cleanStateAfterTest(driver, input, outputRaw, outputSuppressed);
+        }
+    }
+
+    private void cleanStateBeforeTest(final String... topic) throws InterruptedException {
+        for (final String s : topic) {
+            safeDelete(s);
+            CLUSTER.createTopic(s, 1, 1);
+        }
+    }
+
+    private KafkaStreams getCleanStartedStreams(final String appId, final StreamsBuilder builder) {
+        final Properties streamsConfig = mkProperties(mkMap(
+            mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, appId),
+            mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers()),
+            mkEntry(StreamsConfig.POLL_MS_CONFIG, Objects.toString(COMMIT_INTERVAL)),
+            mkEntry(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, Objects.toString(COMMIT_INTERVAL))
+        ));
+        final KafkaStreams driver = new KafkaStreams(builder.build(), streamsConfig);
+        driver.cleanUp();
+        driver.start();
+        return driver;
+    }
+
+    private void safeDelete(final String topic) throws InterruptedException {
+        try {
+            CLUSTER.deleteTopic(topic);
+        } catch (final kafka.common.TopicAlreadyMarkedForDeletionException ignored) {
+            // ignore
+        }
+    }
+
+    private void cleanStateAfterTest(final KafkaStreams driver, final String... topics) throws InterruptedException {
+        driver.cleanUp();
+        for (final String topic : topics) {
+            safeDelete(topic);
+        }
+    }
+
+    private long scaledTime(final long unscaledTime) {
+        return SCALE_FACTOR * unscaledTime;
+    }
+
+    private void produceSynchronously(final String topic, final List<KVT<String, String>> toProduce) {
```

Comment:
Why not use `InegrationTestUtils.produce..` method variants?

Reply:
To improve these tests' readability, I created a simplified "record": `KVT`. This allows us to produce a batch of records, all with potentially different timestamps, at once.

The `IntegrationTestUtils` methods only accept a collection of `KeyValue`, all with the same timestamp, or a collection of `V`, all with the same key and timestamp.

Reply:
I considered adding to the Utils, but I would also have to add KVT. It seems unnecessary at this point to dump a new method and KeyValue-esque class, which are only used in this test, into Util.